folding-schemes/src/frontend/noname/utils.rs

```diff
@@ -0,0 +1,157 @@
+use crate::Error;
+use ark_ff::PrimeField;
+use ark_r1cs_std::fields::fp::FpVar;
+use ark_r1cs_std::R1CSVar;
+use ark_relations::r1cs::SynthesisError;
+use noname::inputs::{parse_inputs, JsonInputs};
+use num_bigint::BigUint;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+pub const NONAME_CIRCUIT_EXTERNAL_INPUTS: &str =
+    "fn main(pub ivc_input: [Field; 2], external_inputs: [Field; 2]) -> [Field; 2] {
+    let xx = external_inputs[0] + ivc_input[0];
+    let yy = external_inputs[1] * ivc_input[1];
+    assert_eq(yy, xx);
+    return [xx, yy];
+}";
+
+pub const NONAME_CIRCUIT_NO_EXTERNAL_INPUTS: &str =
+    "fn main(pub ivc_input: [Field; 2]) -> [Field; 2] {
+    let out = ivc_input[0] * ivc_input[1];
+    return [out, ivc_input[1]];
+}";
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NoNameJSONIVCInput {
+    pub ivc_input: Vec<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NoNameJSONExternalInputs {
+    pub external_inputs: Vec<String>,
+}
+
+impl<F: PrimeField> From<Vec<F>> for NoNameJSONIVCInput {
+    fn from(value: Vec<F>) -> Self {
+        let mut ivc_input = Vec::with_capacity(value.len());
+        for field_element in value {
+            match field_element == F::zero() {
+                true => ivc_input.push("0".to_string()),
+                false => ivc_input.push(field_element.to_string()),
+            }
```

Comment:
Why is this needed? Can't we convert `0` into `String`???

Reply:
I know, that's weird, see [7f19f6b](https://github.com/privacy-scaling-explorations/sonobe/pull/121/commits/7f19f6b802b4dff3b40c47c170fb4c5089565d84)...

Reply:
This is the problem of formatting as String instead of bytes I guess.. Can this be changed? Can we move away from the String type?

Reply:
Nope, since we will then need to format those into a json that can be ingested by the noname witness generator..

Reply:
Nope, we can't move away from the string type, since `Noname` expects field values to be represented as strings when parsing witness inputs, [see](https://github.com/zksecurity/noname/blob/cf9af0b07c248561985fa13107d267655e3619de/src/inputs.rs#L80)

Reply:
That's sad... Thanks mate!